code/modules/vehicles/aircraft.dm

```diff
@@ -0,0 +1,1297 @@
+					y_offset = -weapon_slot_positions[weapon_position_in_list * 2 - 1] + 16
+				if(WEST)
+					x_offset = weapon_slot_positions[weapon_position_in_list * 2]
+					y_offset = -weapon_slot_positions[weapon_position_in_list * 2 - 1] + 32
+				if(NORTHWEST)
+					x_offset = -weapon_slot_positions[weapon_position_in_list * 2 - 1] + 16
+					y_offset = -weapon_slot_positions[weapon_position_in_list * 2 - 1] + 48
+			var/image/gun_overlay = image(gun.icon, gun.icon_state + "_a", pixel_x = x_offset, pixel_y = y_offset, layer = ABOVE_MOB_LAYER-0.01)
+			add_overlay(gun_overlay)
+			weapon_overlays += gun_overlay
+
+/obj/vehicle/sealed/aircraft/examine(mob/user)
+	. = ..()
+	. += "There is [available_passenger_slots() ? "room for [available_passenger_slots()] passengers" : "no room for passengers"]."
+	if(is_driver(user))
+		. += "The fuel gauge reads: <b>[current_fuel]/[max_fuel]</b>"
+	else
+		. += "[pilot ? "It is being piloted by [pilot.name]" : "There is no pilot inside"]."
+	. += "It has <b>[obj_integrity]/[max_integrity] HP</b>"
+	if(attached_weapons)
+		for(var/obj/item/weapon/gun/aircraft/gun AS in attached_weapons)
+			. += "[gun.name] is attached."
+	else
+		. += "No weapon is attached."
+
+/obj/vehicle/sealed/aircraft/attack_hand(mob/living/user)
+	if(is_occupant(user))
+		balloon_alert(user, "Must be outside!")
+		return FALSE
+	if(CHECK_BITFIELD(flags_pass, FLYING) && !CHECK_MULTIPLE_BITFIELDS(user.flags_pass, FLYING))	//Only another flying unit can reach us!
+		balloon_alert(user, "Out of reach!")
+		return
+	if(LAZYLEN(occupants) && user.a_intent == INTENT_GRAB)	//For kicking someone out
+		var/mob/evicted = LAZYLEN(occupants) == 1 ? occupants[1] : tgui_input_list(user, "Choose who to remove from [src]", "Occupant Eviction", occupants)
+		if(!evicted)
+			return
+		visible_message(span_warning("[user.name] is attempting to remove [evicted] from [src]!"))	//Chat message for logging purposes
+		if(!do_after(user, enter_delay * 2, TRUE, user))
+			return
+		return mob_try_exit(evicted)
+	if(deployable)
+		return deploy(user)
+	return ..()
+
+/obj/vehicle/sealed/aircraft/attackby(obj/item/I, mob/user, params)
+	if(is_occupant(user))
+		balloon_alert(user, "Must be outside!")
+		return FALSE
+	if(CHECK_BITFIELD(flags_pass, FLYING) && !CHECK_MULTIPLE_BITFIELDS(user.flags_pass, FLYING))	//Only another flying unit can reach us!
+		balloon_alert(user, "Out of reach!")
+		return
+	. = ..()
+	if(istype(I, /obj/item/reagent_containers/jerrycan))
+		return refuel(I, user)
+	if(istype(I, /obj/item/weapon/gun/aircraft))
+		return attach_weapon(I, user)
+	if(istype(I, /obj/item/ammo_magazine))
+		return reload(I, user)
+
+/obj/vehicle/sealed/aircraft/attack_alien(mob/living/carbon/xenomorph/X, damage_amount, damage_type, damage_flag, effects, armor_penetration, isrightclick)
+	if(CHECK_BITFIELD(flags_pass, FLYING) && !CHECK_MULTIPLE_BITFIELDS(X.flags_pass, FLYING))	//Only another flying beno can hit us!
```

Comment:
?? why are you using 2 different defines / use the operators please

Reply:
CHECK_MULTIPLE_BITFIELDS is needed to make sure all bitfields are present. If not used, mobs with a single flag in flags_pass (like Runners with their table pass flag) can interact with the vehicle while flying.

Reply:
this reply is unrelated to the request